examples/srv6-sighup-flow/src/main.rs

```diff
@@ -115,8 +118,46 @@ where
 
     let pipelines: Vec<_> = ports
         .iter()
-        .map(|port| nf(ReceiveBatch::new(port.clone()), sched).send(port.clone()))
+        .map(|port| {
+            ReceiveBatch::new(port.clone())
+                .map(|p| {
+                    let eth = p.parse::<Ethernet>()?;
+                    let v6 = eth.parse::<Ipv6>()?;
+                    Ok(v6)
+                })
+                .filter(|p| match p.next_header() {
+                    ProtocolNumbers::Ipv6Route => true,
+                    ProtocolNumbers::Tcp => true,
+                    _ => false,
+                })
+                .for_each(|_p| {
+                    warn!(
+                        "Settings/Configuration Static State Val: {:?}",
+                        ATOM_CONF.get()
+                    );
+                    Ok(())
+                })
+                .group_by(
+                    |p| p.next_header(),
+                    |groups| {
+                        compose!(groups,
+                                 ProtocolNumbers::Ipv6Route => |group| {
+                                     group.map(|p| {
+                                         let srh = p.parse::<SegmentRouting<Ipv6>>()?;
+                                         warn!("SR {}", srh);
+                                         Ok(srh.deparse())
+                                     })
+                                 }
+                                 // _ => |group| {
+                                 //     group.for_each(|_p| Ok(()))
+                                 // }
```

Comment:
Yeah the difficulty is the look up later. Essentially you want to be able to override the default catch-all behavior of dropping all the unmatched packets with a different catch-all arm. Not that obvious how with the hashmap. To hack around this, we can fall back to using an artificial discriminator like `1, 2, 3...` in the selector to kinda regroup.

Reply:
@drunkirishcoder did take the `Option` approach, but still hidden from the user. see what you think.